src/Serialization/main.jl

```diff
@@ -70,98 +160,49 @@ function registerSerializationType(ex::Any,
       # ambiguities in their encodings.
 
       serialize_with_id(obj::T) where T <: $ex = $uses_id 
-      serialize_with_id(T::Type{<:$ex}) = $uses_id 
+      serialize_with_id(T::Type{<:$ex}) = $uses_id
+      serialize_with_params(T::Type{<:$ex}) = $uses_params
     end)
 end
 
-macro registerSerializationType(ex::Any, str::Union{String,Nothing} = nothing)
-  return registerSerializationType(ex, false, str)
-end
+macro register_serialization_type(ex::Any, args...)
+  uses_id = false
+  uses_params = false
+  str = nothing
+  for el in args
+    if el isa String
+      str = el
+    elseif el == Symbol("uses_id")
+      uses_id = true
+    elseif el == Symbol("uses_params")
```

Comment:
More idiomatic would be this:
```suggestion
    elseif el == :uses_id
      uses_id = true
    elseif el == :uses_params
```